storage/buffer.go

```diff
@@ -78,12 +81,17 @@ func (b *BufferedSeriesIterator) Seek(t int64) bool {
 // Next advances the iterator to the next element.
 func (b *BufferedSeriesIterator) Next() bool {
 	// Add current element to buffer before advancing.
-	b.buf.add(b.it.At())
+	if !b.done {
+		b.buf.add(b.it.At())
+	}
 
 	ok := b.it.Next()
 	if ok {
 		b.lastTime, _ = b.Values()
```

Comment:
Umm, nope, so if `Next()` returns false, that means the value it held before `Next()` was called is the last. `Next()` either did not advance it or advanced it to an invalid value. 

Even if we set it, it would be most likely the same previous `b.lastTime`.